languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -18536,6 +18536,218 @@ USA
+                <token>better</token>
+                <token>look</token>
+            </antipattern>
+            <antipattern><!-- time to look forward -->
+                <token>time</token>
+                <token>to</token>
+                <token>look</token>
+            </antipattern>
+            <pattern>
+                <marker>
+                    <token inflected="yes">look</token>
+                    <token>forward</token>
+                </marker>
+                <token negate="yes">to
+                    <exception regexp="yes">[,:;()&lt;&gt;&amp;]|&apostrophe;|&quote;</exception><!-- looking forward, -->
+                    <exception regexp="yes">to--.*</exception><!-- 'to' tokenized with 2 hyphens (en dash) -->
+                </token>
+            </pattern>
+            <message>The phrasal verb '\1 \2' (meaning 'anticipate') is usually followed by 'to' and an indirect object.</message>
+            <!-- INCORRECT -->
+            <example correction="">I <marker>look forward</marker>.</example>
+            <example correction="">I'm <marker>looking forward</marker>.</example>
+            <example correction="">I am <marker>looking forward</marker>!</example>
+            <example correction="">that I am <marker>looking forward</marker> for:</example>
+            <example correction="">I <marker>look forward</marker> your response.</example>
+            <example correction="">I'll be <marker>looking forward</marker>.</example>
+            <example correction="">I'll be <marker>looking forward</marker> in hearing from you</example>
+            <example correction=""><marker>Looking forward</marker> into it.</example>
+            <example correction=""><marker>Looking forward</marker> the new release.</example>
+            <example correction=""><marker>Looking forward</marker> too 9.02</example>
+            <example correction=""><marker>Looking forward</marker> for this feature.</example>
+            <example correction=""><marker>Looking forward</marker> from you.</example>
+            <example correction="">I <marker>look forward</marker> hearing from you shortly.</example>
+            <example correction="">Be <marker>looking forward</marker> for next year</example>
+            <example correction="">the one I am most <marker>looking forward</marker> t</example>
+            <example correction="">Steven mentioned a Brazilian restaurant he went to and I am <marker>looking forward</marker> going there</example>
+            <example correction="">It's the only event I'm <marker>looking forward</marker>.</example>
+            <example correction="">Thanks, <marker>looking forward</marker> for your response.</example>
+            <example correction="">My first milestone that I am <marker>looking forward</marker> too is 135x5.</example>
+            <example correction="">linking it here for reference (and <marker>looking forward</marker> for more info</example>
```

Comment:
In this case, "for" precedes a good candidate for an object. The chunk `B-PP` might be useful. In those cases, we can replace the wrong preposition.

Reply:
One last thing: please put AP for the more specific patterns in the general subrule.